src/PostgREST/DbStructure.hs

```diff
@@ -634,22 +642,45 @@ tablesSqlQuery getAll pgVer =
       )|]
     relIsPartition = if pgVer >= pgVersion100 then " AND not c.relispartition " else mempty
 
-allM2ORels :: PgVersion -> Bool -> SQL.Statement () [Relationship]
-allM2ORels pgVer =
+
+-- | Gets many-to-one relationships and one-to-one(O2O) relationships, which are a refinement of the many-to-one's
+allM2OandO2ORels :: PgVersion -> Bool -> SQL.Statement () [Relationship]
+allM2OandO2ORels pgVer =
```

Comment:
Just to comment that this query runs on pg 15 beta 3 and builds correctly. Did not check the code, though.